### PR TITLE
ci: start parachain using data from live network

### PR DESCRIPTION
**Pull request by CertainLach**
```shell
cd docker/forking
docker build ../.. -t parachain-fork -f Dockerfile-parachain-live-fork
docker run -it -v ./launch-config.json:/polkadot-launch/launch-config.json -v ./fork.jsonnet:/polkadot-launch/fork.jsonnet -p 9944:9944 parachain-fork
```

### CODE REVIEW

**Comment by lzadjsf** [comment]
Would you be so kind to move files from docker sub-directoty to earlier introduced .docker sub-directory?
It would solve merge conflict and fits into structure introduced into Github Workflow.

**Comment by CertainLach** [comment]
There was no `.docker` subdirectory before this PR. Moved and resolved conflicts with develop

File: .docker/forking/launch-config.json [hunk 1]
```diff
@@ -0,0 +1,128 @@
+{
+    "relaychain": {
+        "bin": "../polkadot/target/release/polkadot",
+        "chain": "rococo-local",
```
Comment (lzadjsf) [inline .docker/forking/launch-config.json:4]: westend-local

File: .docker/forking/launch-config.json [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                            "validation_upgrade_delay": 1
+                        }
+                    }
+                }
+            }
+        }
+    },
+    "parachains": [
+        {
+            "bin": "../unique-chain/target/release/unique-collator",
```
Comment (lzadjsf) [inline .docker/forking/launch-config.json:87]: full path instead of relative

File: .docker/forking/launch-config.json [hunk 3]
```diff
@@ -0,0 +1,128 @@
+{
+    "relaychain": {
+        "bin": "../polkadot/target/release/polkadot",
```
Comment (lzadjsf) [inline .docker/forking/launch-config.json:3]: Full path instead of relative.

File: .docker/forking/Dockerfile-parachain-live-fork [hunk 4]
```diff
@@ -0,0 +1,101 @@
+# ===== Rust builder =====
+FROM phusion/baseimage:focal-1.1.0 as rust-builder
+LABEL maintainer="Unique.Network"
+
+ARG RUST_TOOLCHAIN=nightly-2022-05-11
```
Comment (lzadjsf) [inline .docker/forking/Dockerfile-parachain-live-fork:5]: To remove explicitly meaning of build ARG.

File: .docker/forking/Dockerfile-parachain-live-fork [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+ARG UNIQUE_BRANCH=develop
```
Comment (lzadjsf) [inline .docker/forking/Dockerfile-parachain-live-fork:6]: To remove explicitly meaning of build ARG.

File: .docker/forking/Dockerfile-parachain-live-fork [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+FROM rust-builder as builder
+
+RUN mkdir unique_parachain
+WORKDIR /unique_parachain
+
+COPY --from=chef /unique_parachain/recipe.json recipe.json
+RUN cargo chef cook "--release" --recipe-path recipe.json
+
+COPY . .
+ARG RUNTIME=quartz
```
Comment (lzadjsf) [inline .docker/forking/Dockerfile-parachain-live-fork:48]: To remove explicitly meaning of build ARG.

Comment (lzadjsf) [reply .docker/forking/Dockerfile-parachain-live-fork:48]: Have to be passed through docker-compose.yml from CI workflow

File: .docker/forking/Dockerfile-parachain-live-fork (hunk 4 above)
Comment (lzadjsf) [reply .docker/forking/Dockerfile-parachain-live-fork:5]: Have to be passed through docker-compose.yml from CI workflow

File: .docker/forking/Dockerfile-parachain-live-fork [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+RUN cargo build --release "--features=$RUNTIME-runtime"
+
+# ===== BUILD POLKADOT =====
+FROM rust-builder as builder-polkadot
+
+RUN mkdir unique_parachain
+WORKDIR /unique_parachain
+
+ARG POLKA_VERSION=release-v0.9.24
```
Comment (lzadjsf) [inline .docker/forking/Dockerfile-parachain-live-fork:57]: To remove explicitly meaning of build ARG.
Have to be passed through docker-compose.yml from CI workflow
Please use definition POLKADOT_BUILD_VERSION from .env in root of the project.

File: .docker/forking/Dockerfile-parachain-live-fork [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+ARG RUST_TOOLCHAIN=nightly-2022-05-11
+ARG UNIQUE_BRANCH=develop
+
+ENV RUST_TOOLCHAIN $RUST_TOOLCHAIN
+ENV UNIQUE_BRANCH $UNIQUE_BRANCH
+
+ENV CARGO_HOME="/cargo-home"
+ENV PATH="/cargo-home/bin:$PATH"
+
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain none
```
Comment (lzadjsf) [inline .docker/forking/Dockerfile-parachain-live-fork:14]: Have to be moved down to string 19. Curl installation required first.

File: .docker/forking/Dockerfile-parachain-live-fork [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+
+RUN apt-get update && \
+    apt-get dist-upgrade -y -o Dpkg::Options::="--force-confold" && \
+    apt-get install -y cmake pkg-config libssl-dev git clang
```
Comment (lzadjsf) [inline .docker/forking/Dockerfile-parachain-live-fork:18]: TO add curl installation

File: .docker/forking/Dockerfile-parachain-live-fork [hunk 10]
```diff
@@ -0,0 +1,101 @@
+# ===== Rust builder =====
+FROM phusion/baseimage:focal-1.1.0 as rust-builder
```
Comment (lzadjsf) [inline .docker/forking/Dockerfile-parachain-live-fork:2]: base imaged decided to use: ubuntu:20.04

File: .docker/forking/Dockerfile-parachain-live-fork [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+    rustup target list --installed && \
+    rustup show
+RUN rustup target add wasm32-unknown-unknown --toolchain $RUST_TOOLCHAIN
+
+RUN cargo install cargo-chef
+
+RUN mkdir unique_parachain
+WORKDIR /unique_parachain
+
+# ===== Chef =====
```
Comment (lzadjsf) [inline .docker/forking/Dockerfile-parachain-live-fork:32]: TO get rid cargo-chef. Only clean build without cache.

File: .docker/forking/Dockerfile-parachain-live-fork [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+
+RUN mkdir chainql
+WORKDIR /chainql
+
+RUN git clone --depth 1 https://github.com/CertainLach/chainql.git . && \
+    cargo build --release
+
+# ===== RUN ======
+
+FROM phusion/baseimage:focal-1.1.0
```
Comment (lzadjsf) [inline .docker/forking/Dockerfile-parachain-live-fork:73]: base image: ubuntu 20.04

**Comment by lzadjsf** [comment]
Доработки забрали в https://github.com/UniqueNetwork/unique-chain/pull/489

**Pull request closed without merging**